server/battleships/domain/src/main/java/eu/optas/domain/ShotResult.java

```diff
@@ -0,0 +1,20 @@
+package eu.optas.domain;
+
+import eu.optas.utils.GameState;
+
+public class ShotResult {
+    private final int[][] grid;
+    private Ship ship;
```

Comment:
Čia spėju final nuėmei, kad nepyktų tas antras konstruktorius, kuriame Ship nėra. Pas mus BE tiems cases, kur value gali nebūti naudojamas Optional type. Naudodamas jį galėsi trinti lauk antrą konstruktorių ir Ship vietoje pasiduoti null.